game.py

```diff
@@ -75,12 +83,50 @@ def check_events(self):
                 if event.key == pygame.K_RIGHT:
                     self.RIGHT_KEY = False
 
+    def asteroids_update(self):
+        for a in self.asteroids:  # Move the asteroids
+            a.move_asteroid()
+            if a.remove:  # Delete asteroid
+                self.asteroids.pop(self.asteroids.index(a))
+            else:
+                if self.collision_rocket(a, self.rocket):
+                    self.asteroids.pop(self.asteroids.index(a))
+                    self.rocket.life -= 1
+                    print("Rocket lives: ", self.rocket.life)
+                else:
+                    for p in self.projectiles:
+                        if self.collision_projectile(a, p):
+                            self.asteroids.pop(self.asteroids.index(a))
+                            self.projectiles.pop(self.projectiles.index(p))
+                            break
+
     def projectiles_update(self):
         for p in self.projectiles:  # Move the projectiles
             p.move_projectile()
             if p.remove:  # Delete projectiles
                 self.projectiles.pop(self.projectiles.index(p))
 
+    def collision_projectile(self, asteroid, projectile):
```

Comment:
I would also make projectiles part of the rocket by adding it as a parameter. This could be done later because it may require refactor.

Reply:
I agree, that would be better, but can we maybe do that when we implement the passengers and hookshot, so I have a more concrete example to implement that? I need to read more on how to do refactors...